lib/src/intl_suggestors/utils.dart

```diff
@@ -0,0 +1,346 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/token.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/dart/element/type.dart';
+import 'package:file/file.dart';
+import 'package:over_react_codemod/src/util/component_usage.dart';
+import 'package:collection/collection.dart' show IterableExtension;
+
+import 'constants.dart';
+
+// -- Node Validation --
+bool isValidStringInterpolationNode(AstNode node) {
+  if (!(node is StringInterpolation)) return false;
+  //We do not need to localize single values.  This should be handled by the
+  // variable being passed in.  IE: ..foo = '$bar' becomes [', '$bar', ']
+  if (node.elements.length == 3 &&
+      node.elements.first.toString() == node.elements.last.toString())
```

Comment:
It seems like this case should be also checking that the elements are empty strings

Reply:
Oh, I meant in this case that `node.elements.first` and `node.elements.last` have empty `.content`. You're right that the `toString()` representations are "'" though.



For example, for the string interpolation `'$bar'`
```
.elements[0]:
  runtimeType: InterpolationStringImpl
  "toString()":  "'"
  "InterpolationString.value": ""

.elements[1]:
  runtimeType: InterpolationExpressionImpl
  "toString()":  "$bar"

.elements[2]:
  runtimeType: InterpolationStringImpl
  "toString()":  "'"
  "InterpolationString.value": ""
```

<details>
<summary>(source code for that ^)</summary>

```dart
import 'package:analyzer/dart/analysis/utilities.dart';
import 'package:analyzer/dart/ast/ast.dart';
import 'package:analyzer/dart/ast/visitor.dart';

void main() {
  final unit = parseString(content: r'''
    foo() => '$bar';
  ''').unit;

  unit.accept(MyVisitor());
}

class MyVisitor extends RecursiveAstVisitor<void> {
  @override
  void visitStringInterpolation(StringInterpolation node) {
    for (var i = 0; i < node.elements.length; i++) {
      final element = node.elements[i];
      print('.elements[$i]:');
      print('  runtimeType: ${element.runtimeType}');
      print('  "toString()":  "$element"');
      if (element is InterpolationString) {
        print('  "InterpolationString.value": "${element.value}"');
      }
      print('');
    }
  }
}
```

</details>

So, I was thinking this check would be more precise if it were:
```dart
  if (node.elements.length == 3) {
    final first = node.elements.first;
    final last = node.elements.last;
    if (first is InterpolationString && last is InterpolationString && 
        first.value.isEmpty && last.value.isEmpty) {
      ...
    }
  }
```